catalog/app/utils/json-schema.spec.js

```diff
@@ -0,0 +1,264 @@
+import { makeSchemaValidator } from './json-schema'
+
+import * as booleansNulls from '../components/JsonEditor/mocks/booleans-nulls'
+import * as compound from '../components/JsonEditor/mocks/compound'
+import * as deeplyNestedArray from '../components/JsonEditor/mocks/deeply-nested-array'
+import * as incorrect from '../components/JsonEditor/mocks/incorrect'
+import * as regular from '../components/JsonEditor/mocks/regular'
+
+describe('utils/json-schema', () => {
+  describe('makeSchemaValidator', () => {
+    it('should return error, when keyword is required but is not provided', () => {
+      const validate = makeSchemaValidator(regular.schema)
+
+      const obj = { optList: [{ id: 1, name: 'Name' }], optEnum: 'one' }
+      const errors = validate(obj)
+      expect(errors).toHaveLength(1)
+      expect(errors[0]).toMatchObject({
+        keyword: 'required',
+        params: { missingProperty: 'a' },
+      })
+    })
+
+    it('should throw an error when schema is incorrect', () => {
+      const validate = makeSchemaValidator(incorrect.schema)
+      const errors = validate({})
+
+      expect(errors).toHaveLength(1)
+      expect(errors[0]).toBeInstanceOf(Error)
+      expect(errors[0].message).toMatch(/schema is invalid/)
+    })
+
+    describe('validator for Schema with enum types', () => {
```

Comment:
tbh i dont think these tests are useful, bc they cover the logic from the third-party code (`ajv`), and it already has its test suite for that. what we really should be testing here is our code (and `describe('makeSchemaValidator', ...)` block does that)

Reply:
These tests give me confidence. Also, it's documentation on particular features we use.
I'm not insisting on preserving them, you can remove these tests in another PR.